lib/cdn-site-hosting/cdn-site-hosting-construct.ts

```diff
@@ -120,20 +116,47 @@ export class CdnSiteHostingConstruct extends cdk.Construct {
 
           const nameOrIndex = name ? name : `${index}`;
 
-          new s3deploy.BucketDeployment(this, `CustomDeploy${nameOrIndex}`, {
-            cacheControl,
-            sources: sources,
-            prune: isSingleDeploymentStep,
-            destinationBucket: this.s3Bucket,
-            distribution: isInvalidationRequired
-              ? this.cloudfrontWebDistribution
-              : undefined,
-            distributionPaths: isInvalidationRequired
-              ? distributionPathsToInvalidate
-              : undefined,
-          });
+          return new s3deploy.BucketDeployment(
+            this,
+            `CustomDeploy${nameOrIndex}`,
+            {
+              cacheControl,
+              sources: sources,
+              prune: isSingleDeploymentStep,
+              destinationBucket: this.s3Bucket,
+              distribution: isInvalidationRequired
+                ? this.cloudfrontWebDistribution
+                : undefined,
+              distributionPaths: isInvalidationRequired
+                ? distributionPathsToInvalidate
+                : undefined,
+            }
+          );
         }
       );
+
+      // For routed SPAs, we need to wait for all other assets to sync
+      // before updating and invalidating the websiteIndexDocument
+      if (props.isRoutedSpa && props.websiteIndexDocumentAsset) {
+        const websiteIndexDocumentDeployment = new s3deploy.BucketDeployment(
+          this,
+          "CustomDeployWebsiteIndexDocument",
+          {
+            cacheControl: [
+              s3deploy.CacheControl.noCache(),
+              s3deploy.CacheControl.maxAge(cdk.Duration.seconds(0)),
+            ],
+            sources: [props.websiteIndexDocumentAsset],
+            prune: false,
+            destinationBucket: this.s3Bucket,
+            distribution: this.cloudfrontWebDistribution,
+            distributionPaths: [`/${props.websiteIndexDocument}`],
+          }
+        );
+        deployments.forEach((deployment) => {
+          websiteIndexDocumentDeployment.node.addDependency(deployment);
+        });
+      }
     } else if (props.sources) {
```

Comment:
Question: do we need to do the same if `props.sources` is provided? What are the respective use cases for these props?